Definition of keycloak-wildfly-adapter-dist in keycloak-adapter-bom doesn't work

### Describe the bug

keycloak-wildfly-adapter-dist is a zip-file artifact. The definition in keycloak-adapter-bom doesn't set the type, so it defaults to jar.

### Version

16.1.0

### Expected behavior

keycloak-adapter-bom probably shouldn't include keycloak-wildfly-adapter-dist. It's a zip-file artifact, I don't see why one would include it as a dependency. But maybe I just don't know the use case?

Alternatively, keycloak-wildfly-adapter-dist could be included with <type>zip</zip> to make it work.

### Actual behavior

Error:
```
[ERROR]   The project com.example:keycloak-bom-dist:1.0.0-SNAPSHOT (...) has 1 error
[ERROR]     'dependencies.dependency.version' for org.keycloak:keycloak-wildfly-adapter-dist:zip is missing. @ line 29, colum
n 21
[ERROR]
```

### How to Reproduce?

Run `mvn package` on the following POM:
```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
>
    <modelVersion>4.0.0</modelVersion>

    <groupId>com.example</groupId>
    <artifactId>keycloak-bom-dist</artifactId>
    <version>1.0.0-SNAPSHOT</version>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>8</maven.compiler.source>
        <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.keycloak.bom</groupId>
                <artifactId>keycloak-adapter-bom</artifactId>
                <version>16.1.0</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-wildfly-adapter-dist</artifactId>
<!--             <version>16.1.0</version> -->
            <type>zip</type>
        </dependency>
    </dependencies>
</project>
```


### Anything else?

keycloak-wildfly-adapter-dist in keycloak-adapter-bom:
https://github.com/keycloak/keycloak/blob/16.1.0/boms/adapter/pom.xml#L52

--- a/boms/adapter/pom.xml
+++ b/boms/adapter/pom.xml
@@ -51,11 +51,6 @@
             </dependency>
             <dependency>
                 <groupId>org.keycloak</groupId>
-                <artifactId>keycloak-wildfly-adapter-dist</artifactId>
-                <version>${project.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>org.keycloak</groupId>
                 <artifactId>keycloak-saml-adapter-core</artifactId>
                 <version>${project.version}</version>
             </dependency>

